slidechain/postpegout.go

```diff
@@ -31,10 +32,13 @@ func (c *Custodian) doPostPegOut(ctx context.Context, assetXDR, anchor, txid []b
 		Anchor:   anchor,
 		Pubkey:   pubkey,
 	}
-	refdata, err := json.Marshal(ref)
+	checkRefdata, err := json.Marshal(ref)
 	if err != nil {
 		return errors.Wrap(err, "marshaling reference data")
 	}
+	if !bytes.Equal(refdata, checkRefdata) {
```

Comment:
This has the same problem as described in #187 : namely, that we're relying on bytewise equality between two different calls to `json.Marshal`, and we might not get it.

The right fix is is to _unmarshal_ the old refdata string into a `pegOut` structure and make sure all the fields are as we expect.

Reply:
> make sure all the fields are as we expect

Sorry, that's misstated. There's nothing to check. We can just rely on the values unmarshaled from the json string for constructing the contract snapshot for `input`, and do away with the now-redundant other fields of the `exports` table.